Homework 08.11.2024 first commit

HKV203 opened the pull request.

Commit:
Homework 08.11.2024 first commit

Commit:
Updated tests

Alexmansar commented on src/main/java/com/andersen/bookstore/controller/Controller.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    private void readProperties() {
+        Properties properties = new Properties();
+        try (InputStream fis = Controller.class.getClassLoader().getResourceAsStream("application.properties")) {
+            properties.load(fis);
+            filepath = properties.getProperty("filepath");
+            availabilityLock = Boolean.parseBoolean(properties.getProperty("availabilityLock"));
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private List<Order> loadLastState(String filepath) {
+        List<Order> orders = new LinkedList<>();
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        try {
+            Scanner scanner = new Scanner(new File(filepath));
+            if (scanner.hasNextLine()) {
+                orders = objectMapper.readValue(scanner.nextLine(), new TypeReference<>() {
+                });
+            }
+            scanner.close();
+        } catch (JsonProcessingException | FileNotFoundException e) {
+            System.out.println(e.getMessage());
+        }
+        return orders;
+    }
+
+    private void saveLastState(Bookstore bookstore, String filepath) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        try {
+            File file = new File(filepath);
+            objectMapper.writeValue(file, bookstore.getOrders());
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
```
I would move it to anotherclass

Alexmansar commented on src/main/java/com/andersen/bookstore/view/Menu.java:
```diff
@@ -17,32 +17,37 @@ public Menu(Bookstore bookstore) {
-            showOrderMenu();
-        } else if (choice == 2) {
-            showOrdersList();
+        switch (choice) {
+            case 1:
+                showOrderMenu();
+                break;
+            case 2:
+                showOrdersList();
+                break;
+            case 3:
+                showAvailabilityMenu();
+                break;
+            default:
+                break;
+
```
use new style please

HKV203 replied:
Does it mean to replace the switch-case construction? If so then with what?

Alexmansar replied:
https://www.baeldung.com/java-switch#1-the-new-switch-expression

Alexmansar reviewed (changes requested):
@Mopgonpblr 
Hi)) in general good job)) I have some comments)so please, check them)) And add please new tests for new logic)

Commit:
1. New tests added
2. readProperties(), loadLastState() and saveLastState() were moved to a new class
3. switch-case style was replaced with the new one

Commit:
Test fix

Alexmansar reviewed (approved):
@Mopgonpblr Please, merge it)) good job)
